python/src/sentencepiece/__init__.py

```diff
@@ -1,1224 +1,29 @@
-
-      if status:
-        self._SetProtoField('add_dummy_prefix', add_dummy_prefix)
-        self._SetProtoField('escape_whitespaces', escape_whitespaces)
-        self._SetProtoField('remove_extra_whitespaces', remove_extra_whitespaces)
-
-    def Normalize(self, input, with_offsets=None):
-      def _normalize(text):
-        if with_offsets:
-          return self._NormalizeWithOffsets(text)
-        return self._Normalize(text)
-
-      if type(input) is list:
-        return [_normalize(x) for x in input]
-      return _normalize(input)
-
-
-    def __getstate__(self):
-      return self.serialized_model_proto()
-
-
-    def __setstate__(self, serialized_model_proto):
-      self.__init__()
-      self.LoadFromSerializedProto(serialized_model_proto)
-
-
-# Register SentencePieceNormalizer in _sentencepiece:
-_sentencepiece.SentencePieceNormalizer_swigregister(SentencePieceNormalizer)
-
-
-import re
-import csv
-import sys
-import os
-from io import StringIO
-from io import BytesIO
-
-
-def _add_snake_case(classname):
-  """Added snake_cased method from CammelCased method."""
-
-  snake_map = {}
-  for k, v in classname.__dict__.items():
-    if re.match(r'^[A-Z]+', k):
-      snake = re.sub(r'(?<!^)(?=[A-Z])', '_',
-                     k).lower().replace('n_best', 'nbest')
-      snake_map[snake] = v
-  for k, v in snake_map.items():
-    setattr(classname, k, v)
-
-
-def _batchnize(classname, name):
-  """Enables batch request for the method classname.name."""
-  func = getattr(classname, name, None)
-  def _func(v, n):
-    if type(n) is int and (n < 0 or n >= v.piece_size()):
-      raise IndexError('piece id is out of range.')
-    return func(v, n)
-
-  def _batched_func(self, arg):
-    if type(arg) is list:
-      return [_func(self, n) for n in arg]
-    else:
-      return _func(self, arg)
-
-  setattr(classname, name, _batched_func)
-
-
-_sentencepiece_processor_init_native = SentencePieceProcessor.__init__
-_sentencepiece_normalizer_init_native = SentencePieceNormalizer.__init__
-setattr(SentencePieceProcessor, '__init__', SentencePieceProcessor.Init)
-setattr(SentencePieceNormalizer, '__init__', SentencePieceNormalizer.Init)
-
-SentencePieceProcessor.Tokenize = SentencePieceProcessor.Encode
-SentencePieceProcessor.Detokenize = SentencePieceProcessor.Decode
-
-for m in [
-    'PieceToId', 'IdToPiece', 'GetScore', 'IsUnknown', 'IsControl', 'IsUnused',
-    'IsByte'
-]:
-  _batchnize(SentencePieceProcessor, m)
-
-_add_snake_case(SentencePieceProcessor)
-_add_snake_case(SentencePieceTrainer)
-_add_snake_case(SentencePieceNormalizer)
-set_random_generator_seed = SetRandomGeneratorSeed
-set_min_log_level = SetMinLogLevel
-
+from .sentencepiece import (
+    ImmutableNBestSentencePieceText,
+    ImmutableSentencePieceText,
+    ImmutableSentencePieceText_ImmutableSentencePiece,
+    SentencePieceNormalizer,
+    SentencePieceProcessor,
+    SentencePieceTrainer,
+    SetMinLogLevel,
+    SetRandomGeneratorSeed,
```

Comment:
Not sure if all of these are meant to be public API. @taku910 should know better.